Lib/email/feedparser.py

```diff
@@ -329,10 +606,11 @@ def _parsegen(self):
             # this onto the input stream until we've scanned past the
             # preamble.
             separator = '--' + boundary
-            def boundarymatch(line):
-                if not line.startswith(separator):
+            def boundarymatch(line, pos = 0, endpos = sys.maxsize):
+                if not line.startswith(separator, pos, endpos):
                     return None
-                return boundaryendRE.match(line, len(separator))
+                return boundaryendRE.match(line, pos + len(separator), endpos)
+            boundarymatch.is_boundary_match = True
```

Comment:
If you want to combine a spilt status var with a function. I prefer we need to use a class instead of it.